error 	addr := conn.LocalAddr().String() //:796

with this error. 
go get github.com/cloudwego/hertz@develop

```
server.WithOnAccept(func(conn network.Conn) context.Context {
	addr := conn.LocalAddr().String() //:796
	log.Printf("BADIP CHECK = %s\n", addr)
	//if isBadIp(addr) {
	//	conn.Close()
	//}
	return context.Background()
}),
```				
https://github.com/cloudwego/hertz/issues/479
```
2023/01/03 19:34:32.569672 engine.go:389: [Info] HERTZ: Using network library=standard
2023/01/03 19:34:32.570042 transport.go:65: [Info] HERTZ: HERTZ: HTTP server listening on address=[::]:443
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xf6a7de]

goroutine 10 [running]:
main.main.func6({0x0?, 0x0?})
        /go/src/example.com/cloud/app/frwll/http2/frwll10.go:796 +0x1e
github.com/cloudwego/hertz/pkg/network/standard.(*transport).serve(0xc000185a00)
        /root/go/pkg/mod/github.com/cloudwego/hertz@v0.4.3-0.20230103082611-81332e6a707d/pkg/network/standard/transport.go:76 +0x2cf
github.com/cloudwego/hertz/pkg/network/standard.(*transport).ListenAndServe(0x1c29f40?, 0x120693e?)
        /root/go/pkg/mod/github.com/cloudwego/hertz@v0.4.3-0.20230103082611-81332e6a707d/pkg/network/standard/transport.go:94 +0x33
github.com/cloudwego/hertz/pkg/route.(*Engine).listenAndServe(0xc0001518c0)
        /root/go/pkg/mod/github.com/cloudwego/hertz@v0.4.3-0.20230103082611-81332e6a707d/pkg/route/engine.go:390 +0x122
github.com/cloudwego/hertz/pkg/route.(*Engine).Run(0xc0001518c0)
        /root/go/pkg/mod/github.com/cloudwego/hertz@v0.4.3-0.20230103082611-81332e6a707d/pkg/route/engine.go:357 +0x16a
github.com/cloudwego/hertz/pkg/app/server.(*Hertz).Spin.func1()
        /root/go/pkg/mod/github.com/cloudwego/hertz@v0.4.3-0.20230103082611-81332e6a707d/pkg/app/server/hertz.go:61 +0x29
created by github.com/cloudwego/hertz/pkg/app/server.(*Hertz).Spin
        /root/go/pkg/mod/github.com/cloudwego/hertz@v0.4.3-0.20230103082611-81332e6a707d/pkg/app/server/hertz.go:60 +0xca
child 1 exited with error: exit status 2
```

--- a/pkg/app/server/hertz_test.go
+++ b/pkg/app/server/hertz_test.go
@@ -803,7 +803,7 @@ func TestOnprepare(t *testing.T) {
 	assert.DeepEqual(t, "the server closed connection before returning the first response byte. Make sure the server returns 'Connection: close' response header before closing the connection", err.Error())
 
 	h = New(
-		WithOnAccept(func(conn network.Conn) context.Context {
+		WithOnAccept(func(conn net.Conn) context.Context {
 			conn.Close()
 			return context.Background()
 		}),
@@ -817,4 +817,18 @@ func TestOnprepare(t *testing.T) {
 	if err == nil {
 		t.Fatalf("err should not be nil")
 	}
+
+	h = New(
+		WithOnAccept(func(conn net.Conn) context.Context {
+			assert.DeepEqual(t, conn.LocalAddr().String(), "127.0.0.1:9231")
+			return context.Background()
+		}),
+		WithHostPorts("localhost:9231"),
+		WithTransport(standard.NewTransporter))
+	h.GET("/ping", func(ctx context.Context, c *app.RequestContext) {
+		c.JSON(consts.StatusOK, utils.H{"ping": "pong"})
+	})
+	go h.Spin()
+	time.Sleep(time.Second)
+	c.Get(context.Background(), nil, "http://127.0.0.1:9231/ping")
 }

--- a/pkg/app/server/option.go
+++ b/pkg/app/server/option.go
@@ -333,7 +333,7 @@ func WithDisablePrintRoute(b bool) config.Option {
 
 // WithOnAccept sets the callback function when a new connection is accepted but cannot
 // receive data in netpoll. In go net, it will be called before converting tls connection
-func WithOnAccept(fn func(conn network.Conn) context.Context) config.Option {
+func WithOnAccept(fn func(conn net.Conn) context.Context) config.Option {
 	return config.Option{F: func(o *config.Options) {
 		o.OnAccept = fn
 	}}

--- a/pkg/common/config/option.go
+++ b/pkg/common/config/option.go
@@ -83,7 +83,7 @@ type Options struct {
 	// If you'd like to check whether the peer IP is in the blacklist, you can use OnAccept.
 	// In go net, OnAccept is executed after connection accepted but before establishing
 	// tls connection. OnConnect is executed after establishing tls connection.
-	OnAccept  func(conn network.Conn) context.Context
+	OnAccept  func(conn net.Conn) context.Context
 	OnConnect func(ctx context.Context, conn network.Conn) context.Context
 
 	// Registry is used for service registry.

--- a/pkg/network/netpoll/transport.go
+++ b/pkg/network/netpoll/transport.go
@@ -41,7 +41,7 @@ type transporter struct {
 	listener         net.Listener
 	eventLoop        netpoll.EventLoop
 	listenConfig     *net.ListenConfig
-	OnAccept         func(conn network.Conn) context.Context
+	OnAccept         func(conn net.Conn) context.Context
 	OnConnect        func(ctx context.Context, conn network.Conn) context.Context
 }
 

--- a/pkg/network/standard/transport.go
+++ b/pkg/network/standard/transport.go
@@ -46,7 +46,7 @@ type transport struct {
 	tls              *tls.Config
 	listenConfig     *net.ListenConfig
 	lock             sync.Mutex
-	OnAccept         func(conn network.Conn) context.Context
+	OnAccept         func(conn net.Conn) context.Context
 	OnConnect        func(ctx context.Context, conn network.Conn) context.Context
 }
 
@@ -73,7 +73,7 @@ func (t *transport) serve() (err error) {
 		}
 
 		if t.OnAccept != nil {
-			ctx = t.OnAccept(c)
+			ctx = t.OnAccept(conn)
 		}
 
 		if t.tls != nil {

